app/models/user.rb

```diff
@@ -1,7 +1,14 @@
 class User < ActiveRecord::Base
   has_many :rides
+  has_many :passengers
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  def passenger?(ride)
+    Ride.find(ride.id).passengers.inject(false) do |memo, passenger|
```

Comment:
You are passing in the ride already so you probably could use `ride.passengers` instead.  This would remove an unnecessary query.

Also I might use `one?` (an enumerable method) instead of inject here.

This might be better on the ride model itself.  `has_passenger?(user)`. Just a thought.